gecko-media/gecko/glue/SourceBuffer.cpp

```diff
@@ -6,20 +6,50 @@
 
 #include "SourceBuffer.h"
 
-namespace mozilla
-{
-namespace dom
-{
+#include "GeckoMediaSource.h"
+#include "MediaSource.h"
+#include "MediaSourceDemuxer.h"
+#include "mozilla/Logging.h"
+
+extern mozilla::LogModule* GetMediaSourceLog();
 
-SourceBuffer::SourceBuffer(GeckoMediaSourceBufferImpl aImpl)
-    : mImpl(aImpl)
+#define MSE_DEBUG(arg, ...)                                                    \
+  MOZ_LOG(GetMediaSourceLog(),                                                 \
+          mozilla::LogLevel::Debug,                                            \
+          ("SourceBuffer(%p)::%s: " arg,                                       \
+           this,                                                               \
+           __func__,                                                           \
+           ##__VA_ARGS__))
+
+namespace mozilla {
+namespace dom {
+
+SourceBuffer::SourceBuffer(GeckoMediaSourceBufferImpl aImpl,
+                           size_t aParentId,
+                           const char* aMimeType)
+  : mImpl(aImpl)
 {
+  mozilla::Maybe<MediaContainerType> mime = MakeMediaContainerType(aMimeType);
```

Comment:
The corresponding upstream code is:

https://searchfox.org/mozilla-central/rev/ba2b0cf4d16711d37d4bf4d267b187c9a27f6638/dom/media/mediasource/MediaSource.cpp#254

Servo will need to be able to reject the promise returned by the JS API MediaSource.addSourceBuffer(mime) with a NotSupportedError when this MakeMediaContainerType() line in GeckoMedia fails. A number of times over the years we've had partners report receiving this error in Firefox, so I think it's a good idea to hook it up.

We don't have to do this now, but we should think about how to facilitate it now, so that we can add that functionality in future. (In general, I/we have been pretty lousy at reporting errors so far in GeckoMedia...)

Reply:
Good point. This is already handled in Servo [here](https://github.com/ferjm/servo/blob/34584d9b163c51f872b72a46673f3687d0938fc6/components/script/dom/mediasource.rs#L259) before even trying to create the SourceBuffer. But I agree that it would not hurt to also return an error if GeckoMedia's MakeMediaContainerType() fails. And yes, we should review error reporting in GeckoMedia at some point.